File: frontend/src/api/listings.ts
// Assigned to: Mai Komar
// Phase: 2 (F2.1)
//
// TODO: Listings API functions.
//
// import client from "./client"
// import { Listing, PaginatedListings } from "../types"
//
// getListings(params: { search?, category?, status?, page?, limit? }): Promise<PaginatedListings>
//   - GET /listings with query params
//   - Return response.data
//
// getListing(id: number): Promise<Listing>
//   - GET /listings/{id}
//   - Return response.data
//
// createListing(formData: FormData): Promise<Listing>
//   - POST /listings with FormData (multipart/form-data)
//   - Must set Content-Type header to "multipart/form-data"
//   - FormData should include: title, description, price, category, and images[] files
//   - Return response.data
//
// updateListing(id: number, data: { title?, description?, price?, status? }): Promise<Listing>
//   - PATCH /listings/{id} with JSON body
//   - Return response.data

import client from "./client"

export const getListings = (params: { search?: string; category?: string; status?: string; page?: number; limit?: number }) => {
    return client.get("/listings", { params }).then((response) => response.data)
}

export const getListing = (id: number) => {
    return client.get(`/listings/${id}`).then((response) => response.data)
}

export const createListing = (formData: FormData) => {
    return client.post("/listings", formData, {
        headers: {
            "Content-Type": "multipart/form-data"
        }
    }).then((response) => response.data)
}

export const updateListing = (id: number, data: { title?: string; description?: string; price?: number; status?: string }) => {
    return client.patch(`/listings/${id}`, data).then((response) => response.data)
}

File: backend/app/routers/listings.py
# Assigned to: Lakshan Kandeepan
# Phase: 2 (B2.5)
#
# TODO: Implement listings API endpoints.
#
# Create router = APIRouter()
#
# GET /
#   - Query params: search (str), category (str), status (str, default="active"),
#     page (int, default=1), limit (int, default=20)
#   - Auth: Optional (public endpoint)
#   - Call listing_service.get_listings() with filters
#   - Return 200: PaginatedListings {listings[], total, page, limit}
#   - Phase 3 (B3.5): Add min_price, max_price query params for filtering
#
# POST /
#   - Auth: Required (Depends(get_current_user))
#   - Accept multipart/form-data:
#       - title: str = Form(...)
#       - description: Optional[str] = Form(None)
#       - price: float = Form(...)
#       - category: Optional[str] = Form(None)
#       - images: List[UploadFile] = File(default=[])
#   - Call listing_service.create_listing() which:
#       - Creates Listing in DB
#       - Saves each image to uploads/ with a UUID filename
#       - Creates Image records in DB
#   - Return 201: ListingOut
#
# GET /{listing_id}
#   - Auth: Optional
#   - Fetch listing by ID with eager-loaded images and seller
#   - Return 404 if not found
#   - Return 200: ListingOut
#
# PATCH /{listing_id}
#   - Auth: Required (must be the listing owner)
#   - Request body: ListingUpdate (JSON)
#   - Only the listing's seller can update it → 403 if not owner
#   - Update only the provided fields
#   - Return 200: ListingOut

from fastapi import APIRouter, Depends, status
from app.routers.messages import router as messages_router

router = APIRouter()

router.include_router(
    messages_router,
    prefix="/{listing_id}/messages",
    tags=["Messages"],
)

@router.get("/health", status_code=status.HTTP_200_OK)
def listing_health():
    return {"message": "Hello, World! From listings.py"}